### ISSUE TITLE
Any plan to support github.com/containers/storage based container runtime?

### ISSUE THREAD

**Issue by wjiangjay** [wagoodman/dive#172]
For example, crio

**Comment by stobias123** [comment]
+1 would love support for podman

**Comment by wagoodman** [comment]
Indeed, this is the plan before 1.0.0 release.

**Comment by nc-ruth** [comment]
@wagoodman do you have a status for this?
Or are you missing details etc.? 

I would love to see Dive work with Podman 🥇

**Comment by wagoodman** [comment]
TLDR; status hasn't changed and I'm missing lots of details :) .

@nc-ruth I've been a bit too busy to get to this, however, it is still something I really want to get into dive (when I find a bit more time). I haven't done enough research to find the best way to integrate multiple runtime engines. Essentially, this would mean exporting image filesystems by layer, similar to the `docker save` command. This can be done by querying the runtime engine or the registries directly --there are different idioms for different engines. I get a feeling that the current interface for image parsing and analysis will need to change as I learn more about theses engines and their various export formats; I took an initial stab at separating out the logic to allow for more formats and processors. 

One of the outstanding problems is the ergonomics of the CLI: how might one simply change the engine or source? Initially it seems that having a switch like `--source <value>` where value can be `docker`, `crio`, `podman`, etc, however, it seems a bit clunky. Maybe something akin to a connection string would suffice? (e.g. `docker://<image>` such as `docker://ubuntu:16.04` or `crio://<image>`, etc). I'm still simmering on how someone would easily swap between sources without confusing folks about where the image coming from.

I'm also looking forward to working on this --I'll get to learn a lot more about podman/crio/containerd/*. (I also welcome any help for those who have the time/expertise!)

**Comment by wagoodman** [comment]
For folks following along here, checkout the latest v0.9.0 release... there is initial podman support. I have not attempted to run dive within a container and mount the necessary libpod directories, so I'd recommend using dive as a binary on your system if you're trying it out (though, if someone here is willing to figure that out, I'd welcome a README update :)
